### DISCUSSION TITLE
After the network configuration fails, ipxe cannot ping through any ip address, even 127.0.0.1

### DISCUSSION THREAD

**Discussion by IAseven** [ipxe/ipxe#1204] [General]
I have two physical servers of the same model and bios version on two networks. After I configure the network after ipxe is started, one server works properly while the other server cannot ping any ip address. Even if ping 127.0.0.1 is used, a timeout message is displayed.
The ip and network configuration is correct when I confirm, and the network configuration is normal when I start centos7 on the machine in the network environment in question, but the network configuration is wrong only in the ipxe environment.
Here is my script for configuring the network in ipxe:

```
#!ipxe
echo set
ifopen net0 \
&& vcreate --tag 1045 net0 \
&& set net0-1045/ip 10.230.45.118 \
&& set net0-1045/gateway 10.230.45.254 \
&& set net0-1045/netmask 255.255.255.0 \
&& sleep 1
echo ***********************************
echo vcreate --tag 1045 net0
echo set net0-1045/ip 10.230.45.118
echo set net0-1045/gateway 10.230.45.254
echo set net0-1045/netmask 255.255.255.0
echo ***********************************

echo show xxxx
show net0-1045/ip
show net0-1045/gateway
show net0-1045/netmask

ping --count 3 10.230.45.118 || shell
```

The following is a screenshot of the NIC of my physical machine：
![图片](https://github.com/ipxe/ipxe/assets/38178039/de722e76-75f7-43db-b6a5-e5e1e600fea3)

ipxe ping Error message
<img width="693" alt="1714290268860" src="https://github.com/ipxe/ipxe/assets/38178039/ac2d45c9-bd08-4274-ba6c-9deb63d0f1e7">
<img width="1217" alt="1714290313735" src="https://github.com/ipxe/ipxe/assets/38178039/09edf37b-d935-423c-afe3-e82cfddd3910">
![图片](https://github.com/ipxe/ipxe/assets/38178039/4dd74009-3228-4aae-8dd3-c891a263602f)
![图片](https://github.com/ipxe/ipxe/assets/38178039/d66665f8-cb8e-4b6c-a67a-ea1f97dce0fa)

I would like to know how to use debug=xxx in make to troubleshoot this problem.

Thank you very much for your opinion

**Comment by NiKiZe** [comment]
Since we don't have enough information yet to know if this is an issue with iPXE I moved it to discussions.

You have 2 servers, are these identical?
What does ifstat show on these 2 machines after configuration?
How do you start iPXE?

You seem to already have debugs enabled?

**Reply by IAseven** [reply]
Yes, I added some debug options, but I'm not sure if I added the correct ones,
I use isolinux and iso images to start ipxe 

My two servers are of the same model and version, but in two different network environments

**Reply by IAseven** [reply]
These are the vlans and ip addresses configured on eno1 nics after centos7 is installed on the physical machine.
PING is normal
![图片](https://github.com/ipxe/ipxe/assets/38178039/4be782f7-eb4e-4224-91f1-99a62df68bc6)

**Reply by NiKiZe** [reply]
Your tests in iPXE is pinging localhost, does that differ between working and non working? If not that isn't really a relevant test.

If you have network issues, consider tcpdump to check where packets are dropped or similar.

**Reply by IAseven** [reply]
When I ping 127.0.0.1 in ipxe, the same error is displayed. In principle, ping 127.0.0.1 should not go through the NIC

**Reply by mcb30** [reply]
Please note that iPXE does not set up any kind of localhost route, and 127.0.0.1 has no meaning to iPXE. It will simply treat it as a normal external IPv4 address and try to send packets out on to the network.

Pinging a local interface's IP address will similarly not do what you expect. 

Since iPXE does not have support for listening sockets, we don't bother to include the extra logic that would be required to route packets back to our own local addresses, and any tests involving pinging a local or loopback address will not give any meaningful result.

**Reply by IAseven** [reply]
Thank you very much for your answer. May I ask, in my case, which debug parameters should be specified in make to better troubleshoot this problem?

**Reply by NiKiZe** [reply]
I'm not sure we know what your "problem" is?
What is your end goal? What do you want to do in iPXE? What happens if you try to do any kind of download? Do you get any errors? And have you checked what the server sees when you try to make a download?

Rather than iPXE, I would debug network port configuration and traffic, as well as server traffic.

**Comment by IAseven** [comment]
Thank you very much, the cause of the problem has been found. The parameters configured on the upstream switch cause a delay in arp learning. You must wait 1 minute to learn the mac of the network card. After I changed the configuration of the switch, it is normal
